internal/k8s/status_test.go

```diff
@@ -15,6 +18,171 @@ import (
 	"k8s.io/client-go/tools/cache"
 )
 
+func TestUpdateTransportServerStatus(t *testing.T) {
+	ts := &conf_v1alpha1.TransportServer{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:      "ts-1",
+			Namespace: "default",
+		},
+		Status: conf_v1alpha1.TransportServerStatus{
+			State:   "before status",
+			Reason:  "before reason",
+			Message: "before message",
+		},
+	}
+
+	fakeClient := fake_v1alpha1.NewSimpleClientset(
+		&conf_v1alpha1.TransportServerList{
+			Items: []conf_v1alpha1.TransportServer{
+				*ts,
+			}})
+
+	tsLister := cache.NewStore(cache.DeletionHandlingMetaNamespaceKeyFunc)
+
+	err := tsLister.Add(ts)
+	if err != nil {
+		t.Errorf("Error adding TransportServer to the transportserver lister: %v", err)
+	}
+	su := statusUpdater{
+		transportServerLister: tsLister,
+		confClient:            fakeClient,
+		keyFunc:               cache.DeletionHandlingMetaNamespaceKeyFunc,
+		externalListenerEndpoints: []conf_v1alpha1.ExternalEndpoint{
+			{
+				IP:   "123.123.123.123",
+				Port: "1234",
+			},
+		},
+	}
+
+	err = su.UpdateTransportServerStatus(ts, "after status", "after reason", "after message")
+	if err != nil {
+		t.Errorf("error updating transportserver status: %v", err)
+	}
+	updatedTs, _ := fakeClient.K8sV1alpha1().TransportServers(ts.Namespace).Get(context.TODO(), ts.Name, meta_v1.GetOptions{})
+
+	expectedStatus := conf_v1alpha1.TransportServerStatus{
+		State:             "after status",
+		Reason:            "after reason",
+		Message:           "after message",
+		ExternalEndpoints: []conf_v1alpha1.ExternalEndpoint{{IP: "123.123.123.123", Port: "1234"}},
+	}
+
+	if diff := cmp.Diff(expectedStatus, updatedTs.Status); diff != "" {
+		t.Errorf("Unexpected status (-want +got):\n%s", diff)
+	}
+}
+
+func TestUpdateTransportServerStatusIgnoreNoChange(t *testing.T) {
+	ts := &conf_v1alpha1.TransportServer{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:      "ts-1",
+			Namespace: "default",
+		},
+		Status: conf_v1alpha1.TransportServerStatus{
+			State:   "same status",
+			Reason:  "same reason",
+			Message: "same message",
+		},
+	}
+
+	fakeClient := fake_v1alpha1.NewSimpleClientset(
+		&conf_v1alpha1.TransportServerList{
+			Items: []conf_v1alpha1.TransportServer{
+				*ts,
+			}})
+
+	tsLister, _ := cache.NewInformer(
```

Comment:
would it make sense to also simplify here and TestUpdateTransportServerStatusMissingTransportServer ? 
```
tsLister := cache.NewStore(cache.DeletionHandlingMetaNamespaceKeyFunc)
```